### PR TITLE
Refactor `ComponentViewItem` to use native class

### PR DESCRIPTION
**Pull request by chancancode**

### CODE REVIEW

File: app/controllers/component-tree.js [hunk 1]
```diff
@@ -137,7 +136,7 @@ export default Controller.extend({
     viewArray.forEach(viewItem => {
       let cachedExpansion = expandedStateCache[getIdFromObj(viewItem)];
       if (cachedExpansion !== undefined) {
-        viewItem.set('expanded', cachedExpansion);
```
Comment (rwwagner90) [inline app/controllers/component-tree.js]: We not longer need `set` here?

File: app/controllers/component-tree.js [hunk 2]
```diff
@@ -210,7 +209,7 @@ export default Controller.extend({
     this.filteredArray.forEach((item) => {
       const id = getIdFromObj(item);
       if (objects.includes(id)) {
-        item.set('expanded', state);
```
Comment (rwwagner90) [inline app/controllers/component-tree.js]: Or here?

File: app/controllers/component-tree.js (hunk 1 above)
Comment (nummi) [reply app/controllers/component-tree.js]: The class is using @tracked.

**Comment by rwwagner90** [comment]
@chancancode this seems fine to me, we just need to fix the tests.

**Comment by chancancode** [comment]
Yeah! I looked into it and I know why it's not working. It needs a bigger refactor, I'll get back to this next week.

**Comment by chancancode** [comment]
I’ll resubmit this as part of a bigger refactor to use the new debug render tree api. I was hoping that this can be separated out relatively cleanly, but it needed more work that start to eat into the other refactor so probably doesn’t make sense to do it in isolation anymore.

**Pull request closed without merging**